pkg/manager/member/pd_member_manager.go

```diff
@@ -566,9 +566,44 @@ func getNewPDSetForTidbCluster(tc *v1alpha1.TidbCluster) (*apps.StatefulSet, err
 		}
 	}
 
+	env := []corev1.EnvVar{
+		{
+			Name: "NAMESPACE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.namespace",
+				},
+			},
+		},
+		{
+			Name:  "PEER_SERVICE_NAME",
+			Value: controller.PDPeerMemberName(tcName),
+		},
+		{
+			Name:  "SERVICE_NAME",
+			Value: controller.PDMemberName(tcName),
+		},
+		{
+			Name:  "SET_NAME",
+			Value: setName,
+		},
+		{
+			Name:  "TZ",
+			Value: tc.Spec.Timezone,
+		},
+	}
+
 	dnsPolicy := corev1.DNSClusterFirst // same as k8s defaults
 	if tc.Spec.PD.HostNetwork {
 		dnsPolicy = corev1.DNSClusterFirstWithHostNet
+		env = append(env, corev1.EnvVar{
+			Name: "POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.name",
+				},
+			},
+		})
```

Comment:
same as https://github.com/pingcap/tidb-operator/pull/1202 to make it compatible with v1.0.4